pyam/core.py

```diff
@@ -1565,10 +1539,79 @@ def map_regions(self, map_col, agg=None, copy_col=None, fname=None,
         if not inplace:
             return ret
 
+    def subtract(self, other, axis='variable', new_name=None,
+                 ignore_meta_conflict=False):
+        """
+        Subtract data in ``other`` from ``self``
+
+        Parameters
+        ----------
+        other : pyam.IamDataFrame
+            Object containing timeseries data to subtract
+
+        axis : str, default ``variable``
+            Column to use when subtracting the two sets of data
+            (e.g. ``variable``)
+
+        new_name : str, default None
+            Name of the value to use in the ``axis`` column (e.g., if
+            ``axis='variable'``, the new variable name)
+
+        ignore_meta_conflict : bool, default False
+            If False and ``other`` is an IamDataFrame, raise an error if
+            any meta columns present in ``self`` and ``other`` are not
+            identical.
+
+        Raises
+        ------
+        ValueError
+            The metadata columns in ``self`` and ``other`` are not identical
+
+        ValueError
+            ``self`` or ``other`` contains more than value for ``join_col``
+
+        NotImplementedError
+            The type of ``other`` is not yet supported
+        """
+        op = BinaryOp(self, other, ignore_meta_conflict)
+        data, meta = op.calc(subtract_op, axis, new_name)
+        ret = IamDataFrame(data)
+        for col in meta:
```

Comment:
why can this be not just `self.meta = meta`? Like done here:
https://github.com/IAMconsortium/pyam/blob/7d46b2c1c3a46d799c1b09dbabc9dbe88e80ee07/pyam/core.py#L1020

Reply:
done!